var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -540,9 +540,17 @@ def cmake_args(self):
             projects.append("polly")
             cmake_args.append(define("LINK_POLLY_INTO_TOOLS", True))
 
-        cmake_args.append(from_variant("BUILD_SHARED_LIBS", "shared_libs"))
-        cmake_args.append(from_variant("LLVM_BUILD_LLVM_DYLIB", "llvm_dylib"))
-        cmake_args.append(from_variant("LLVM_USE_SPLIT_DWARF", "split_dwarf"))
+        cmake_args.extend([
+            from_variant("BUILD_SHARED_LIBS", "shared_libs"),
+            from_variant("LLVM_BUILD_LLVM_DYLIB", "llvm_dylib"),
+            from_variant("LLVM_LINK_LLVM_DYLIB", 'llvm_dylib'),
```

Comment:
Can we please make this a separate variant? I regularly build the dylib but want to build a statically linked clang, since clang and family run ~15% slower when linked dynamically.  We should have this option, I just don't want it to be the only way to get a dylib.

Reply:
should we just follow the naming scheme of llvm cmake vars closer? `+build_llvm_dylib` `+link_llvm_dylib`?

Reply:
Done that with a 'when' clause. This is breaking, but I think breaking changes to packages are fine.